MAUI/DataForm/Editors.md

```diff
@@ -0,0 +1,961 @@
+
+N> Class cannot be directly set as data type for autocomplete editor in this complex type scenario.
+
+{% tabs %}
+{% highlight MainPage.xaml.cs %}
+
+    dataForm.ItemsSourceProvider = new DataFormItemsSourceProvider();
+    dataForm.DataObject = new DataModel();
+    dataForm.RegisterEditor("City", DataFormEditorType.AutoComplete);
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+ 
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null && e.DataFormItem.FieldName == "City" && e.DataFormItem is DataFormAutoCompleteItem autoCompleteItem)
+        {
+            autoCompleteItem.DisplayMemberPath = "City";
+            autoCompleteItem.SelectedValuePath = "PostalCode";
+        }
+    } 
+ 
+    public class DataFormItemsSourceProvider : IDataFormSourceProvider
+    {
+         public object GetSource(string sourceName)
+        {
+            if (sourceName == "City")
+            {
+                List<Address> details = new List<Address>();
+                details.Add(new Address() { City = "Colorado Springs", PostalCode = 1 });
+                details.Add(new Address() { City = "Chicago", PostalCode = 2 });
+                details.Add(new Address() { City = "Columbus", PostalCode = 3 });
+                details.Add(new Address() { City = "Portland", PostalCode = 4 });
+                details.Add(new Address() { City = "Paris", PostalCode = 5 });
+                details.Add(new Address() { City = "Las Vegas", PostalCode = 6 });         
+                details.Add(new Address() { City = "New York", PostalCode = 7 });
+                details.Add(new Address() { City = "Cincinnati", PostalCode = 8 });
+                details.Add(new Address() { City = "San Diego", PostalCode = 9 });
+
+                return details;
+            }
+            return new List<string>();
+        }
+    }
+
+    public class DataModel
+    {
+        [Display(Name ="First Name")]
+        public String FirstName { get; set; } 
+        public string City { get; set; }
+    }
+
+    public class Address
+    {
+        public int PostalCode { get; set; }
+        public string City { get; set; }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+`Image`
+
+#### AutoComplete editor modes
+
+The [DataFormAutoCompleteEditor]`Link` provides display the filtered suggestions.
+
+The `AutoCompleteMode` property is used to choose the suggestion display mode in the `DataFormAutoCompleteItem` class.
+
+#### AutoComplete editor suggestion options
+
+The phenomenon of string comparison for filtering suggestions can be changed using the "StartsWith" and "Containswith" `Link` property. The default filtering strategy is “StartsWith”, and it is case insensitive. The available filtering modes are,
```

Comment:
Containswith = check this, it is Contains only

Reply:
containswith to contains